src/frequenz/sdk/microgrid/component/_component.py

```diff
@@ -106,13 +108,64 @@ def _component_category_from_protobuf(
     return ComponentCategory(component_category)
 
 
+class ComponentMetadata:
+    """Base class for component metadata classes."""
+
+
+class GridMetadata(ComponentMetadata):
+    """Metadata for a grid connection point."""
+
+    def __init__(self, max_current: float) -> None:
```

Comment:
It'd be nice if this were a dataclass. But changing it to a dataclass turns `Component.metadata` into a dict, causing a few comparison tests to fail. Has anyone here encountered such an issue before?

Reply:
Have you tried with `@dataclass(frozen=True)`?

Reply:
Is there a reason why `max_current` is not of type `Current`?

Reply:
>Is there a reason why max_current is not of type Current?

Because that's how I started, and it did not make sense to change it to `Current` because this gets changed to `Fuse` anyway.

Reply:
If there are no further suggestions, then I'll mark this resolved.

Reply:
Found the culprit: https://github.com/frequenz-floss/frequenz-sdk-python/blob/1fe5cb8c0103f37580988fa03da3a345a3079290/src/frequenz/sdk/microgrid/_graph.py#L476

The graph module is converting components to dicts.

Reply:
Will push changes today to get this as a dataclass.

Reply:
Done. Ready for a look.